packages/cached/test/inputs/persistent_storage_test_input.dart

```diff
@@ -0,0 +1,886 @@
+import 'package:cached_annotation/cached_annotation.dart';
+import 'package:source_gen_test/annotations.dart';
+
+@ShouldGenerate(
+  r'''
+abstract class _$StaticPersistedRepository {}
+
+class _StaticPersistedRepository
+    with StaticPersistedRepository
+    implements _$StaticPersistedRepository {
+  _StaticPersistedRepository() {
+    _init();
+  }
+
+  Future<void> _init() async {
+    if (_isStaticCacheLocked == true) {
+      return _completer.complete();
```

Comment:
```suggestion
      return;
```

It will complete it to early if object will be created 2 times